### ISSUE TITLE
CamelCase rule doesn't account for initial character

### ISSUE THREAD

**Issue by snitin315** [eslint/eslint#14786]
**Tell us about your environment**

* **ESLint Version:** 7.30.0
* **Node Version:** v14.16.0
* **npm Version:** 7.19.1
* **Operating System:**  MacOS

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
/*eslint camelcase: "error"*/
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
/*eslint camelcase: "error"*/

let CamelCase = 10;
let camelCase = 'hello';
let Camel_Case = 10;
```
**What did you expect to happen?**
Expected to report a violation on `let CamelCase = 10` as well.

**What actually happened? Please copy-paste the actual, raw output from ESLint.**

Reporting error only for `let Camel_Case = 10`

**Steps to reproduce this issue:**

[OnlineDemo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgY2FtZWxjYXNlOiBcImVycm9yXCIqL1xuXG5sZXQgQ2FtZWxDYXNlID0gMTA7XG5sZXQgY2FtZWxDYXNlID0gJ2hlbGxvJztcbmxldCBDYW1lbF9DYXNlID0gMTA7XG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjEyLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

**Are you willing to submit a pull request to fix this bug?**
Yes

**Comment by snitin315** [comment]
Probably not a bug as both `camelCase` and `CamelCase` are valid camel case but an enhancement to the rule by adding an option (maybe `lowerCase: true`) to force initial chacter as `lowercase` or `uppercase`.

**Comment by mdjermanovic** [comment]
> Probably not a bug as both `camelCase` and `CamelCase` are valid camel case syntax but an enhancement to the rule by adding an option (something like `lowerCase: true`) to force initial chacter as `lowercase` or `uppercase`.

I'm not sure if this option would be generally useful since codebases often have both (e.g. class names and user-defined components in React usually begin with a capital letter).

**Comment by nzakas** [comment]
This rule is intended to flag IDs in the form `foo_bar` as being invalid. Both `CamelCase` and `camelCase` are considered valid by this rule.

Given that this is just enforcing a style preference, we aren't making any further changes to this rule.

**Comment by snitin315** [comment]
That makes sense, I think we can close it then.
